src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,470 @@
+    /// An instance of this class is created for each service replica by the Service Fabric runtime.
+    /// </summary>
+    internal sealed class DependencyUpdateErrorProcessor : IServiceImplementation
+    {
+        public DependencyUpdateErrorProcessor(
+            IReliableStateManager stateManager,
+            ILogger<DependencyUpdateErrorProcessor> logger,
+            BuildAssetRegistryContext context,
+            IOptions<DependencyUpdateErrorProcessorOptions> options
+            )
+        {
+            _stateManager = stateManager;
+            _context = context;
+            _options = options.Value;
+            Logger = logger;
+        }
+
+        private readonly IReliableStateManager _stateManager;
+
+        private readonly DependencyUpdateErrorProcessorOptions _options;
+
+        private ILogger<DependencyUpdateErrorProcessor> Logger { get; }
+
+        private readonly BuildAssetRegistryContext _context;
+
+        /// captures [marker]: <> (subscriptionId: 'a9f0536e-15cc-4d1f-79a4-08d79acf8bba', method: 'UpdateAssetsAsync', errorMessage: 'testing one')
+        /// from the issue or comment body.
+        private static readonly Regex CommentMarker = new Regex(@"\[marker\]: <> \(subscriptionId: '(?<subscriptionId>[^,]*)', method: '(?<method>[^,]*)'(?:, errorMessage: '(?<errorMessage>[^,]*)')?\)");
+
+        [CronSchedule("0 0 0/1 1/1 * ? *", TimeZones.PST)]
+        public async Task ProcessDependencyUpdateErrorsAsync()
+        {
+            
+            if (_options.ConfigurationRefresherEndPointUri == null && _options.DynamicConfigs == null)
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+                return;
+            }
+            await _options.ConfigurationRefresherEndPointUri.Refresh();
+            if (bool.TryParse(_options.DynamicConfigs["FeatureManagement:DependencyUpdateErrorProcessor"],
+                out var dependencyUpdateErrorProcessorFlag) && dependencyUpdateErrorProcessorFlag)
+            {
+                IReliableDictionary<string, DateTimeOffset> checkpointEvaluator =
+                    await _stateManager.GetOrAddAsync<IReliableDictionary<string, DateTimeOffset>>("checkpointEvaluator");
+                try
+                {
+                    DateTimeOffset checkpoint;
+                    using (ITransaction tx = _stateManager.CreateTransaction())
+                    {
+                         checkpoint = await checkpointEvaluator.GetOrAddAsync(
+                         tx,
+                         "checkpointEvaluator",
+                         DateTimeOffset.UtcNow
+                         );
+                        await tx.CommitAsync();
+                    }
+                    await CheckForErrorsInRepositoryBranchHistoryTableAsync(checkpoint, _options.GithubUrl, checkpointEvaluator);
+                }
+                catch (Exception exe)
```

Comment:
I think you need to do here the same thing you're doing in LN 124. This Exception is catching too many things.

Reply:
I did and a reverted it last week, cos I thought I was adding redundancy (face palm )